crates/forge/bin/cmd/bind.rs

```diff
@@ -174,6 +178,58 @@ impl BindArgs {
                 }
             })
             .collect::<Result<Vec<_>, _>>()?;
+
+        if self.src_only {
+            //We are relying on compilationTarget being present in the metadata
+            //and that it is an object with a single key { "path/to/contract.sol": "ContractName" }
+            //After parsing the metadata, we filter out the contracts that are not in the src directory
+            abigens = abigens
+                .into_iter()
+                .map(|abi| {
+                    // Read abi as string
+                    let abi_str = match abi.source().get() {
+                        Ok(s) => s,
+                        Err(_) => {
+                            return Err(eyre::eyre!(
+                                "Failed to read abi as string for {}",
+                                abi.name()
+                            ))
+                        }
+                    };
+                    // Deserialize abi as JSON
+                    let json = match serde_json::from_str::<serde_json::Value>(&abi_str) {
+                        Ok(json) => json,
+                        Err(_) => {
+                            return Err(eyre::eyre!(
+                                "Failed to deserialize abi as json for {}",
+                                abi.name()
+                            ))
+                        }
+                    };
+                    // Get the compilationTarget
+                    let contract_path = json["metadata"]["settings"]["compilationTarget"]
+                        .as_object()
+                        .map(|obj| obj.keys().collect::<Vec<_>>())
+                        .map(|keys| *keys.first().unwrap()) //compilationTarget is always an object with a single key
+                        .unwrap() //compilationTarget is always present
+                        .clone();
+                    Ok((abi, contract_path))
+                })
+                .collect::<Result<Vec<_>, _>>()?
+                .into_iter()
+                .filter_map(
+                    // Filter out contracts that are not in the src directory
+                    |(abi, contract_path)| {
+                        if contract_path.starts_with("src/") {
```

Comment:
Shouldn't this be pulled from the config as the name of this directory can be configured?